x-pack/plugins/fleet/public/components/agent_enrollment_flyout/instructions.tsx

```diff
@@ -0,0 +1,374 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, useMemo, useEffect } from 'react';
+import { EuiSteps, EuiLink, EuiText, EuiSpacer } from '@elastic/eui';
+import type { EuiContainedStepProps } from '@elastic/eui/src/components/steps/steps';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import { safeDump } from 'js-yaml';
+
+import {
+  useGetOneEnrollmentAPIKey,
+  useLink,
+  useFleetStatus,
+  useGetAgents,
+  useStartServices,
+  useKibanaVersion,
+  sendGetOneAgentPolicyFull,
+  sendGetOneAgentPolicy,
+} from '../../hooks';
+
+import {
+  deploymentModeStep,
+  ServiceTokenStep,
+  FleetServerCommandStep,
+  useFleetServerInstructions,
+  addFleetServerHostStep,
+} from '../../applications/fleet/sections/agents/agent_requirements_page/components';
+import { FleetServerRequirementPage } from '../../applications/fleet/sections/agents/agent_requirements_page';
+
+import { policyHasFleetServer } from '../../applications/fleet/sections/agents/services/has_fleet_server';
+
+import type { FullAgentPolicy } from '../../../common/types/models/agent_policy';
+
+import { FLEET_SERVER_PACKAGE } from '../../constants';
+
+import type { PackagePolicy } from '../../../common';
+
+import { FLEET_KUBERNETES_PACKAGE } from '../../../common';
+import { fullAgentPolicyToYaml, agentPolicyRouteService } from '../../services';
+
+import { StandaloneInstructions } from '../enrollment_instructions/standalone';
+import type { InstructionsType } from '../enrollment_instructions/standalone';
+
+import {
+  InstallationModeSelectionStep,
+  PolicyOrEnrollmentTokenSelection,
+  AgentEnrollmentConfirmationStep,
+  InstallManagedAgentStep,
+  IncomingDataConfirmationStep,
+  InstallStandaloneAgentStep,
+  ConfigureStandaloneAgentStep,
+} from './steps';
+import type { InstructionProps } from './types';
+
+import { usePollingAgentCount } from './confirm_agent_enrollment';
+
+const DefaultMissingRequirements = () => {
+  const { getHref } = useLink();
+
+  return (
+    <>
+      <FormattedMessage
+        id="xpack.fleet.agentEnrollment.agentsNotInitializedText"
+        defaultMessage="Before enrolling agents, {link}."
+        values={{
+          link: (
+            <EuiLink href={getHref('overview')}>
+              <FormattedMessage
+                id="xpack.fleet.agentEnrollment.setUpAgentsLink"
+                defaultMessage="set up central management for Elastic Agents"
+              />
+            </EuiLink>
+          ),
+        }}
+      />
+    </>
+  );
+};
+
+const FleetServerMissingRequirements = () => {
+  return <FleetServerRequirementPage />;
+};
+
+export const Instructions = React.memo<InstructionProps>(
```

Comment:
I think the name `Instructions` is a little too generic. What about `AgentEnrollmentInstructions`?